modules/widgets/src/compass-widget.tsx

```diff
@@ -117,7 +117,7 @@ export class CompassWidget implements Widget<CompassWidgetProps> {
   }
 
   handleCompassReset() {
-    const viewId = this.viewId || 'default-view';
+    const viewId = this.viewId || this.viewport?.id || 'default-view';
```

Comment:
Seems a bit clunky that this needs to be duplicated across widgets, it would be nice if we could push this complexity into the `WidgetManager` somehow. Also, can the `this.viewport?.id` ever be different from the `viewId`? If not, we could just do:

```ts
const viewId = this.viewport!.id || 'default-view';
```

Reply:
Widgets implement how they handle this themselves since not every viewport is tied to a view (FullscreenWidget has no view code in it).

I think viewId and viewport can differ in multi-view. I have some questions about multi-view widgets to look into next, so I'll follow up on this soon.